src/prototypes/draft-email-textarea/js/draft-email-textarea.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * The contents of this function will be inlined in a Gutenberg Custom HTML block
+ */
+export const runProposedInlineJS = () => {
+  // The elements all will be a part of the same Gutenberg Custom HTML block.
+  // I chose to use IDs since the HTML and JS are scoped to the same Gutenberg
+  // block (the JS functionaly won't be shared elsewhere, so I didn't use classes)
+  const copyBtn = document.getElementById('copy-btn');
+  const draftEmailBtn = document.getElementById('draft-email-btn');
+  const draftEl = document.getElementById('email-draft');
+  const copySuccessTooltipEl = document.getElementById('copy-success-tooltip');
+  const copyFailTooltipEl = document.getElementById('copy-fail-tooltip');
+
+  // All the things required to run this feature
+  const requirements = [
+    copyBtn,
+    draftEmailBtn,
+    draftEl,
+    copySuccessTooltipEl,
+    copyFailTooltipEl,
+    navigator,
+    navigator.clipboard,
+    navigator.clipboard.writeText,
+  ];
+  // Handle use case when all requirements are not met
+  if (!requirements.every((requirement) => requirement)) {
+    // Hide the "copy" btn since it will not be functional
+    if (copyBtn) copyBtn.hidden = true;
+    // Exit this script
+    return;
+  }
+
+  // Keep track of the tooltip timeout ID to clear it if the tooltip is
+  // activated again before it is hidden
+  let hideTooltipDelayTimeoutId;
+
+  /**
+   * Handles showing & hiding the "copy" action tooltip
+   * @param {HTMLElement} tooltipEl
+   * @param {Number} [hideTooltipDelay=4000]
+   */
+  const showTooltip = (tooltipEl, hideTooltipDelay = 3000) => {
+    // Unhide it and start the "intro" animation
+    tooltipEl.hidden = false;
+    tooltipEl.classList.add('is-animating-intro');
+    // Clear any existing timeouts
+    if (hideTooltipDelayTimeoutId) clearTimeout(hideTooltipDelayTimeoutId);
+    // Hide the tooltip after a delay
+    hideTooltipDelayTimeoutId = setTimeout(() => {
+      tooltipEl.classList.remove('is-animating-intro');
+      tooltipEl.classList.add('is-animating-outro');
+    }, hideTooltipDelay);
+    // Listen for when the tooltip animation ends
+    tooltipEl.addEventListener('animationend', onTooltipAnimationEnd);
```

Comment:
Just a heads up, if it's possible for the tooltip to contain an element with an animation, this listener could be fired by the tooltip's child element finishing its animation. I don't _think_ that's an issue here, but I wanted to flag it in case it's possible we add animated children in the future, since this can be a tricky issue to debug.